server/container/guice/rabbitmq/src/main/java/org/apache/james/modules/rabbitmq/RabbitMQModule.java

```diff
@@ -104,14 +98,26 @@ protected void configure() {
     }
 
     @Provides
+    @Named(RABBITMQ_CONFIGURATION_NAME)
     @Singleton
-    private RabbitMQConfiguration getMailQueueConfiguration(PropertiesProvider propertiesProvider) throws ConfigurationException {
+    private Configuration getConfiguration(PropertiesProvider propertiesProvider) throws ConfigurationException {
```

Comment:
why do you need to publish this Configuration object ? You could very well call this method in getMailQueueConfiguration instead, don't you think ?

Reply:
Report only one time a missing configuration file?

Anyway, I can duplicate the catch block if you want.

Reply:
Memoize ?

Reply:
Having had a look a memoized supplier needs to access the PropertiesProvider and it makes it hards share it betweens `getMailQueueConfiguration` & `getMailQueueViewConfiguration`

(the memoized supplier could not be a class field).

I would need to add a private class for doing such things, but it would not look more understandable nor concise.

Another alternative would be a 'cache' for that single value. But explicitly managing mutable state here looks bad to me.